### PR TITLE
Force upgrade once app version is under minimum version maintained

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds a version check at start to be able to force the upgrade once we'll release a non-backward compatible version of the app (or API).

**Note**: It also fixes the status bar color on splash and authentication screens.

##### iOS

Default:

Kapture 2019-03-19 at 16 42 49

Upgrade needed:

Kapture 2019-03-19 at 16 43 27

##### Android

Default:

Kapture 2019-03-19 at 16 37 09

Upgrade needed:

Kapture 2019-03-19 at 16 38 11

- [ ] **Breaking changes ?**
- [x] **Extra dependency ?** [`semver-compare`](https://github.com/substack/semver-compare)

### Testing Strategy

- [ ] Already covered by tests
- [ ] Manual testing
- [x] Unit testing

### CODE REVIEW

File: jest.config.unit.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ module.exports = {
       statements: 98
     },
     'src/modules': {
-      branches: 100,
+      branches: 97,
```
Comment (Volubyl) [inline jest.config.unit.js]: umh you're decrease 3% of code coverage directly are you sure about this. I mean 3% is a lot.

File: scripts/pull-translations.js [hunk 2]
```diff
@@ -43,6 +43,9 @@ const fetchTranslations = (
   );
 };
 
+const formatTranslation = (translation: string): string =>
```
Comment (Volubyl) [inline scripts/pull-translations.js]: I don't understand why those update are located in this PR. What's the link btwn translation and app version ?

File: src/components/authentication.js [hunk 3]
```diff
@@ -58,8 +58,11 @@ const styles = StyleSheet.create({
   }
 });
 
+export const TOP_COLOR = BLUE_COORP_DARK;
+export const BOTTOM_COLOR = BLUE_COORP_LIGHT;
+
 const Authentication = ({onPress}: Props) => (
-  <Gradient colors={[BLUE_COORP_DARK, BLUE_COORP_LIGHT]} style={styles.container}>
+  <Gradient colors={[TOP_COLOR, BOTTOM_COLOR]} style={styles.container}>
```
Comment (Volubyl) [inline src/components/authentication.js]: same comment. Maybe you should be more specific in you PR description to give a real overview of what you embed in your PR

File: jest.config.unit.js (hunk 1 above)
Comment (VincentCATILLON) [reply jest.config.unit.js]: I don't know why i don't have 100% branches of my store module. :/ We don't have a lot of files in modules, that's why it decreases of 3%.

**Comment by VincentCATILLON** [comment]
@AdrieanKhisbe no worries :)

**Pull request merged**